### PR TITLE
treewide: remove unnecessary enableParallelBuilding when using cmake, meson, qmake

### PR DESCRIPTION
**Pull request by siraben**
###### Motivation for this change
I'm following the heuristic used by [nixpkgs-hammering](https://github.com/jtojnar/nixpkgs-hammering/blob/e42eede0e3dc0316a003106024146f9db681688d/overlays/unnecessary-parallel-building.nix)

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by siraben** [comment]
Building minimal ISO to assess breakage.

File: pkgs/development/libraries/libtiff/default.nix [hunk 1]
```diff
@@ -38,9 +38,8 @@ stdenv.mkDerivation rec {
 
   buildInputs = [ libdeflate ]; # TODO: move all propagatedBuildInputs to buildInputs.
 
-  enableParallelBuilding = true;
-
-  doCheck = true;
+  doInstallCheck = true;
```
Comment (Mindavi) [inline pkgs/development/libraries/libtiff/default.nix:41]: Does autoconf also build in parallel by default? Or is this a mistake?

**Review by Mindavi — APPROVED** [review]
One comment, otherwise seems to be good. I don't think it's an issue if it's enabled explicitly though.

File: pkgs/development/libraries/libtiff/default.nix (hunk 1 above)
Comment (siraben) [reply pkgs/development/libraries/libtiff/default.nix:41]: Ah, this must have been caused by my rebase from master to staging.

**Comment by siraben** [comment]
ISO builds.

**Pull request merged**